### ISSUE TITLE
How can I use just only chromedriver to start selenium standalone

### ISSUE THREAD

**Issue by wck821829906** [vvo/selenium-standalone#433]
@vvo 
## Expected Behavior
I want only download chrome driver and start selenium standalone with customized config

## Current Behavior
all types of browsers driver are required to download

## Steps to Reproduce
1.  use seleniumConfig.js to control driver infomation
![image](https://user-images.githubusercontent.com/17002975/56564543-d6108f80-65e0-11e9-81fb-af9ceb24bd84.png)

2.  set the driver install script with above seleniumConfig.js 
![image](https://user-images.githubusercontent.com/17002975/56564716-4a4b3300-65e1-11e9-8435-348a151a3a1b.png)

3.  execute script: **npm run installdriver**
expected: only chrome driver is downloaded
actual: all types of drivers are downloaded
![image](https://user-images.githubusercontent.com/17002975/56564968-ce9db600-65e1-11e9-9b6a-8bfb5e6eb6f8.png)

**Comment by jcsmorais** [comment]
I'm also able to reproduce this on version 6.16.0, I'm using the CLI and getting the same outcome:

```sh
selenium-standalone install --version=3.141.59 --drivers.chrome.version=2.46
----------
selenium-standalone installation starting
----------

---
selenium install:
from: https://selenium-release.storage.googleapis.com/3.141/selenium-server-standalone-3.141.59.jar
to: /Users/jcsmorais/project/node_modules/selenium-standalone/.selenium/selenium-server/3.141.59-server.jar
---
chrome install:
from: https://chromedriver.storage.googleapis.com/2.46/chromedriver_mac64.zip
to: /Users/jcsmorais/project/node_modules/selenium-standalone/.selenium/chromedriver/2.46-x64-chromedriver
---
firefox install:
from: https://github.com/mozilla/geckodriver/releases/download/v0.23.0/geckodriver-v0.23.0-macos.tar.gz
to: /Users/jcsmorais/project/node_modules/selenium-standalone/.selenium/geckodriver/0.23.0-x64-geckodriver


selenium-standalone installation [==                  ] 11% 4.8s

-----
selenium-standalone installation finished
```

**Comment by jcsmorais** [comment]
@serbanghita any thoughts on this :arrow_up:?

**Comment by jcsmorais** [comment]
Just figured that there's a flag to install specific drivers only:

```sh
# install a single driver within the default list (chrome, ie, edge, firefox)
selenium-standalone install --singleDriverInstall=chrome
```

@wck821829906 maybe that'll help you too.

**Comment by wck821829906** [comment]
@jcsmorais, Does it support install special selenium version (such as: 3.141.59 ) with above option arguments ?

**Comment by versedi** [comment]
> Just figured that there's a flag to install specific drivers only:
> 
> ```shell
> # install a single driver within the default list (chrome, ie, edge, firefox)
> selenium-standalone install --singleDriverInstall=chrome
> ```
> 
> @wck821829906 maybe that'll help you too.

Installation is successful but you still can't run it with single driver. It fails due to missing `geckodriver`.

Even while trying to run with a config file that specifies only a single driver:

`seleniumConfig.json`

```
{
  "drivers": {
    "chrome": {
      "version": "2.43",
      "arch": "x64",
      "baseURL": "https://chromedriver.storage.googleapis.com"
    }
  }
}

  ```

`./node_modules/.bin/selenium-standalone start --config=tests/seleniumConfig.json`

```
Error: Missing /home/ci/work/app/node_modules/selenium-standalone/.selenium/geckodriver/0.23.0-x64-geckodriver
    at /home/ci/work/app/node_modules/selenium-standalone/lib/check-paths-existence.js:15:20
    at suppressedCallback (fs.js:199:5)
    at FSReqWrap.args [as oncomplete] (fs.js:140:20)
```

But geckodriver can't be installed since it's FreeBSD and it requires a binary built for it (it's available in ports though, see https://www.freshports.org/www/geckodriver/).


It would be really nice to be able to run it with a single driver. 

If I understand correctly this happens due to this code block inside `start.js`

```
  if (opts.drivers) {
    // Merge in missing driver options for those specified
    opts.drivers = mapValues(opts.drivers, function(config, name) {
      return merge({}, defaultConfig.drivers[name], config);
    });
  } else {
    opts.drivers = defaultConfig.drivers;
  }
```

The config is merged with defaultConfig so firefox is being enabled even though it's not specified in the configuration file.

**Comment by kylewhitaker** [comment]
There is a difference between how the command line and API start commands merge drivers custom config with the default config. The command line will persist other drivers from the default config that are not specified in the custom config, whereas the API will ignore default config drivers not specified in a custom config.

See the difference with [this Repl sandbox example](https://repl.it/@kylewhitaker/selenium-standalone-custom-config).

References:
[Command Line source code](https://github.com/vvo/selenium-standalone/blob/master/bin/selenium-standalone#L54)
[API source code](https://github.com/vvo/selenium-standalone/blob/master/lib/start.js#L39-L46)

**Comment by wck821829906** [comment]
my solution is to override default config (/node_modules/selenium-standalone/lib/default-config.js) with my customized config file, just like this:
![image](https://user-images.githubusercontent.com/17002975/68656261-707bf480-056c-11ea-8139-705b77666274.png)

**Comment by kylewhitaker** [comment]
@wck821829906 Overriding the default config within node_modules is not a good solution. Other consumers of the code will have to manually do this. Also, your work could be overwritten with future installs forcing you to complete this task more than once.

I've submitted PR #448 to help solve this issue. In the meantime, you could write a simple javascript file to run the install/start for you. The programmatic version of selenium install/start works the way you want it to; it's the CLI that does not.

**selenium.js**
```js
var selenium = require('selenium-standalone');
var config = require('/path/to/seleniumConfig.js');

selenium.install(config, function (installErr) {
  if (installErr) throw installErr;
  selenium.start(config, function (startErr) {
    if (startErr) throw startErr;
  });
});
```

Then, execute `node selenium.js` instead of using the CLI.

**Comment by vvo** [comment]
closed by #448
